pkg/receive/handler.go

```diff
@@ -193,3 +213,80 @@ func ValidateLabels(logger log.Logger, next http.Handler, labels ...string) http
 		next.ServeHTTP(w, r)
 	}
 }
+
+func (h *Handler) TransformWriteRequest(logger log.Logger, next http.Handler) http.HandlerFunc {
```

Comment:
I am only a tiny bit worried about the fact that we're doing the whole process of `snappy decode -> unmarshall -> filter -> marshall -> snappy encode` after the `ValidateLabels` middleware does the exact same thing. Isn't it too much extra work considering the scale that this service runs at? Would it be possible to avoid the extra work?

Reply:
Yeah, for now, there isn't a whole lot of usage on this endpoint, but would be good to improve and iterate on it! :)